modules/collator-selection/src/lib.rs

```diff
@@ -294,7 +296,7 @@ pub mod pallet {
 			let incoming = CandidateInfo {
 				who: who.clone(),
 				deposit,
-				last_block: frame_system::Pallet::<T>::block_number(),
+				start_session: T::ValidatorSet::session_index() + 2,
```

Comment:
```suggestion
                // pallet-session have 1 session delay so this will be applied on 1 session after next session
                // Tracking issue: https://github.com/paritytech/substrate/issues/8650
				start_session: T::ValidatorSet::session_index() + 2,
```